src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,53 @@
+package seedu.contactbook.data.person;
+
+import seedu.contactbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public static final String MESSAGE_CONTACT_CONSTRAINTS = "Person contactes can be in any format";
+    public static final String CONTACT_VALIDATION_REGEX = ".+";
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        String trimmedContact = contact.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidContact(trimmedContact)) {
+            throw new IllegalValueException(MESSAGE_CONTACT_CONSTRAINTS);
+        }
+        this.value = trimmedContact;
+    }
+
+    /**
+     * Returns true if a given string is a valid contact.
+     */
+    public static boolean isValidContact(String test) {
+        return test.matches(CONTACT_VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Beware of the issue we discussed in tutorial, i.e. subclasses of different type may be "equal" if they contain the same value by this code.